commands/admin/addtime.js

```diff
@@ -19,22 +19,16 @@ module.exports.load = (client) => {
       }
 
       let delta = parseInt(args[1])
-      client.loadUserData(args[0].replace(/[<@!>]/g, ''), async res => {
-        if (res === null) {
-          return client.errorMessage(message, 'User doesn\'t exist in DB')
-        }
+      let userInfo = await client.loadUserData(args[0].replace(/[<@!>]/g, ''))
+      if (userInfo == null) {
+        return client.errorMessage(message, 'User doesn\'t exist in DB')
+      }
 
-        res.current_session_playtime += delta
-        res.overall_session_playtime += delta
-        client.writeUserData(args[0].replace(/[<@!>]/g, ''), res, () => {
-          message.reply('Added time to user.')
-            .then(m => {
-              setTimeout(() => {
-                m.delete()
-              }, client.settings.res_destruct_time * 1000)
-            })
-        })
-      })
+      userInfo.current_session_playtime += delta
+      userInfo.overall_session_playtime += delta
+      await client.writeUserData(args[0].replace(/[<@!>]/g, ''), userInfo)
+      let m = await message.reply('added time to user.')
```

Comment:
We could maintain the old code path which is "faster" to return from the command. This seems like a legit use of `.then`. Also, neither the original nor this re-factor includes proper error handling :-)

Reply:
I think the whole self-destructing message thing should probably be a library method anyway. We use that technique all over the place and we could put the error-handling there.